pkgs/applications/networking/cluster/tektoncd-cli/default.nix

```diff
@@ -13,35 +13,47 @@ buildGoModule rec {
 
   vendorSha256 = null;
 
-  doCheck = false;
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X github.com/tektoncd/cli/pkg/cmd/version.clientVersion=${version}"
+  ];
 
   nativeBuildInputs = [ installShellFiles ];
 
-  buildPhase = ''
-    make bin/tkn
+  preBuild = ''
+    # VENDOR-LICENSE breaks build/check as go files are still included
+    rm -r ./third_party/VENDOR-LICENSE
+  '';
+
+
+  preCheck = ''
+    # Change the golden files to match our desired version
+    sed -i "s/dev/${version}/" pkg/cmd/version/testdata/TestGetVersions-*.golden
   '';
 
   installPhase = ''
-    install bin/tkn -Dt $out/bin
+    # Only install tkn
+    install -D $GOPATH/bin/tkn $out/bin/tkn
```

Comment:
Not sure about the $GOPATH here. Why did you change that?

Reply:
I think we should use `subPackages` and use postInstall to install completions and man pages rather than overriding the installPhase IMO.

Reply:
Well is there an easy way to filter out the relevant subPackages? cmd/tkn obviously but that has none of the tests

Reply:
Ok I'm excluding `third_party` & `cmd/docs` so now I don't need to rm third_party or get rid of cmd/docs result

Reply:
The interface for `excludePackages` is really uncomfortable. It'd be nice if it could take an array like `subPackages`